File: JavaCivitas/src/juegoTexto/Controlador.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package juegoTexto;

import civitas.CivitasJuego;
import civitas.OperacionesJuego;
import civitas.SalidasCarcel;
import civitas.GestionesInmobiliarias;
import civitas.Respuestas;
import civitas.OperacionInmobiliaria;

public class Controlador {
    
    CivitasJuego juego;
    VistaTextual vista;

    Controlador(CivitasJuego juego, VistaTextual vista){
            this.juego=juego;
            this.vista = vista;
    }
    
    void juega() {

        vista.setCivitasJuego(juego);

        while (!juego.finalDelJuego()) {

            vista.actualizarVista();
            vista.pausa();

            OperacionesJuego operacion = juego.siguientePaso();
            vista.mostrarSiguienteOperacion(operacion);

            if (operacion != OperacionesJuego.PASAR_TURNO) {
                vista.mostrarEventos();;
            }

            if (!juego.finalDelJuego()) {
                //metodo comprar

                if ( operacion == OperacionesJuego.COMPRAR ) {

                    if (vista.comprar() == Respuestas.SI) {
                        juego.comprar();
                    }
                    juego.siguientePasoCompletado(operacion);

                } else if (operacion == OperacionesJuego.GESTIONAR) {
                    vista.gestionar();
                    OperacionInmobiliaria ig = new OperacionInmobiliaria(GestionesInmobiliarias.values()[vista.getGestion()],vista.getPropiedad());

                    switch (ig.getGestion()) {
                        case VENDER:
                            juego.vender(vista.getPropiedad());
                            break;
                        case HIPOTECAR:
                            juego.hipotecar(vista.getPropiedad());
                            break;
                        case CANCELAR_HIPOTECA:
                            juego.cancelarHipoteca(vista.getPropiedad());
                            break;
                        case CONSTRUIR_CASA:
                            juego.construirCasa(vista.getPropiedad());
                            break;
                        case CONSTRUIR_HOTEL:
                            juego.construirHotel(vista.getPropiedad());
                            break;
                        case TERMINAR:
                            juego.siguientePasoCompletado(operacion);
                            break;

                    }

                } else if (operacion == OperacionesJuego.SALIR_CARCEL) {
                    SalidasCarcel salida;
                    salida = vista.salirCarcel();
                    if (salida == SalidasCarcel.PAGANDO) {
                        juego.salirCarcelPagando();
                    } else {
                        juego.salirCarcelTirando();
                    }
                    juego.siguientePasoCompletado(OperacionesJuego.SALIR_CARCEL);

                }

                if(juego.finalDelJuego()) {

                    System.out.println("El juego ha finalizado debido a que un jugador ha caido en bancarrota. Ranking: ");
                    for (int i = 0; i < juego.ranking().size(); i++) {

                        System.out.println(juego.ranking().get(i));
                    }
                }
            }
        }
    }
}

File: civitasruby/pkg/Civitas-0.0.1/lib/controlador.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'civitas_juego'
require_relative 'operacion_inmobiliaria'
require_relative 'operaciones_juego'
require_relative 'respuestas'
require_relative 'gestiones_inmobiliarias'
require_relative 'salidas_carcel'
module Civitas

class Controlador
  def initialize(juego,vista)
    @juego=juego
    @vista=vista
  end
  
  def juega
    @vista.set_civitas_juego(@juego)
    while(!@juego.final_del_juego)
      @vista.actualizar_vista
      @vista.pausa
      op=@juego.siguiente_paso
      @vista.mostrar_siguiente_operacion(op)
      if(op!=Civitas::Operaciones_juego::PASAR_TURNO)
        @vista.mostrarEventos
      end
      
      if(!@juego.final_del_juego)
        case op
        when Operaciones_juego::COMPRAR
          res=@vista.comprar
          
          if(res==Civitas::Respuestas::SI)
            @juego.comprar
          end
           @juego.siguiente_paso_completado(op)
           
          
          
        when Operaciones_juego::GESTIONAR
          @vista.gestionar
          @vista.iGestion
          @vista.iPropiedad
          lista=[Civitas::GestionesInmobiliarias::VENDER,Civitas::GestionesInmobiliarias::HIPOTECAR,Civitas::GestionesInmobiliarias::CANCELAR_HIPOTECA,Civitas::GestionesInmobiliarias::CONSTRUIR_CASA,Civitas::GestionesInmobiliarias::CONSTRUIR_HOTEL,Civitas::GestionesInmobiliarias::TERMINAR]
          oi=OperacionInmobiliaria.new(lista[@vista.iGestion], @vista.iPropiedad)
          
          case lista[@vista.iGestion]
          when GestionesInmobiliarias::VENDER
            if(!@juego.get_jugador_actual.get_nombre_propiedades!=nil)
              @juego.vender(oi.numPropiedad)
            end
            
          when GestionesInmobiliarias::HIPOTECAR
            if(!@juego.get_jugador_actual.get_nombre_propiedades!=nil)
              @juego.hipotecar(oi.numPropiedad)
            end
            
          when GestionesInmobiliarias::CANCELAR_HIPOTECA
            if(!@juego.get_jugador_actual.get_nombre_propiedades!=nil)
              @juego.cancelar_hipoteca(oi.numPropiedad)
            end
            
          when GestionesInmobiliarias::CONSTRUIR_CASA
            @juego.construir_casa(oi.numPropiedad)
            
          when GestionesInmobiliarias::CONSTRUIR_HOTEL
            @juego.construir_hotel(oi.numPropiedad)
          else
            @juego.siguiente_paso_completado(op)
          end
            
          
        when   Operaciones_juego::SALIR_CARCEL
          salida=@vista.salir_carcel
          case salida
          when SalidasCarcel::PAGANDO
            @juego.salir_carcel_pagando
          when SalidasCarcel::TIRANDO
            @juego.salir_carcel_tirando
          end
          @juego.siguiente_paso_completado(op)
        end
        
        
      else 
        lista=Array.new
        lista=@juego.ranking
        lista.reverse
        for i in 0...lista.size
          puts lista[i].nombre+"\n"
        end
      end
         
        
         
      
    end
    
    
  end

  
  
  
  end
end